src/services/queryresults.js

```diff
@@ -1563,16 +1563,25 @@ export default new (class QueryResultsService extends G3WObject {
 
     const { query = {} } = this.state;
 
-    const data = 'search' === query.type
-      ? { field: query.search.join() }                                // search results + pagination (see: https://github.com/g3w-suite/g3w-client/pull/743)
-      : { fids: features.filter(f => !(layer.filter || {}).active || f.selection.selected).map(f => f.attributes[G3W_FID]).join(',') }; // other query types ('point', 'polygon', 'bbox' ..)
-    
-    data.down_with_relations = down_with_relations; 
-  
-    //In the case of pdf type need to add html element
-    if ('pdf' === type) {
-      data.html = html;
-    }
+    // filter out undefined properties
+    const data = Object.fromEntries(Object.entries({
+
+      down_with_relations,
+
+      // search results + pagination (see: https://github.com/g3w-suite/g3w-client/pull/743)
+      field: 'search' === query.type && query.search
+        ? query.search.join()
+        : undefined,
+
+      // other query types ('point', 'polygon', 'bbox' ..)
+      fids: 'search' !== query.type || !query.search
+        ? query.fids || features.filter(f => !layer.filter?.active || f.selection.selected).map(f => f.attributes[G3W_FID]).join(',')
+        : undefined,
+
+      // html element (pdf)
+      html: 'pdf' === type ? html : undefined,
+
+    }).filter(([_, v]) => v !== undefined));
```

Comment:
@volterra79 Do you confirm that it is superfluous, ie. it work the same way?

```suggestion
    const data = {

      down_with_relations,

      // search results + pagination (see: https://github.com/g3w-suite/g3w-client/pull/743)
      field: 'search' === query.type && query.search
        ? query.search.join()
        : undefined,

      // other query types ('point', 'polygon', 'bbox' ..)
      fids: 'search' !== query.type || !query.search
        ? query.fids || features.filter(f => !layer.filter?.active || f.selection.selected).map(f => f.attributes[G3W_FID]).join(',')
        : undefined,

      // html element (pdf)
      html: 'pdf' === type ? html : undefined,

    };
```

Reply:
@Raruto ok for me